{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Main methodology results\n",
    "\n",
    "*Disclaimer*: please note that many parts of this code require the preprocessed data from ADNI (both genetic and diagnostic related) as input. This data has not been uploaded to the repository for privacy reasons."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import pandas as pd\n",
    "import datetime, pickle\n",
    "import create_datasets.create_nx_datasets\n",
    "from ml_models.machine_learning_models import create_class_LOAD, baseline_model\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**1. Obtain genes of interest**\n",
    "\n",
    "Using DisGeNET to get Gene-Disease-Associations (GDAs) to Alzheimer's Disease (AD gene set) and other neurodegenerative diseases (ND). This is already obtained from [first part of the results](1_main_methodology.ipynb)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**2. Obtain biological networks**\n",
    "\n",
    "Using genes of interst obtained from DisGeNET, obtain PPI between them from STRING. This is already obtained from [first part of the results](1_main_methodology.ipynb)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**3. Data preprocessing**\n",
    "\n",
    "Please refer to `data_preprocessing` subdirectory for this part.\n",
    "1. [make_BED_files.R](data_preprocessing/make_BED_files.R) creates BED files with the genomic coordinates of the genes of interest. This is already obtained from [first part of the results](1_main_methodology.ipynb).\n",
    "2. [extract_and_annotate_missense_LOAD.sh](data_preprocessing/extract_and_annotate_missense_LOAD.sh) is the script for obtaining missense variants from the VCF files."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**4. Create graph datasets**\n",
    "\n",
    "Create graph datasets (one graph representing each patient) for different targets with ADNI dataset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Input directory: data\n",
      "Output directory: data/graph_datasets/LOAD\n",
      "\n",
      "\n",
      "data/ND_STRING_PPI_edgelist.txt\n",
      "Network used: ND original\n",
      "# nodes = 139\n",
      "# edges = 263\n",
      "\n",
      "Dataset used: LOAD\n",
      "    CHROM        POS          ID REF ALT Allele       Consequence    IMPACT  \\\n",
      "0       1   11854476   rs1801131   A   C      C  missense_variant  MODERATE   \n",
      "1       1   20977000   rs1043424   A   C      C  missense_variant  MODERATE   \n",
      "2       1  209782343  rs11119314   T   C      C  missense_variant  MODERATE   \n",
      "3       2  210558162    rs741006   G   A      A  missense_variant  MODERATE   \n",
      "4       2  234113301      rs9247   C   T      T  missense_variant  MODERATE   \n",
      "5       3   12393125   rs1801282   C   G      G  missense_variant  MODERATE   \n",
      "6       3  133475722   rs1799852   G   A      A  missense_variant  MODERATE   \n",
      "7       3  133494354   rs1049296   C   T      T  missense_variant  MODERATE   \n",
      "8       4   23815662   rs8192678   C   T      T  missense_variant  MODERATE   \n",
      "9       4   23815681  rs17574213   C   T      T  missense_variant  MODERATE   \n",
      "10      5  179264731     rs10277   G   A      A  missense_variant  MODERATE   \n",
      "11      6   26093141   rs1800562   C   T      T  missense_variant  MODERATE   \n",
      "12      6   52849325   rs1802061   G   A      A  missense_variant  MODERATE   \n",
      "13      6  161781225   rs1801334   G   A      A  missense_variant  MODERATE   \n",
      "14      6  161807855   rs1801582   C   G      G  missense_variant  MODERATE   \n",
      "15      7  143095153  rs11768549   G   A      A  missense_variant  MODERATE   \n",
      "16      7  154587793  rs10236495   A   G      G  missense_variant  MODERATE   \n",
      "17      7  154672646   rs3817522   A   G      G  missense_variant  MODERATE   \n",
      "18      9  135139826   rs3739927   T   C      C  missense_variant  MODERATE   \n",
      "19      9  135172412   rs2296869   T   C      C  missense_variant  MODERATE   \n",
      "20      9  135173685   rs2296871   A   G      G  missense_variant  MODERATE   \n",
      "21     11   27679916      rs6265   C   T      T  missense_variant  MODERATE   \n",
      "22     12    7023123    rs710415   A   G      G  missense_variant  MODERATE   \n",
      "23     12   40631791  rs10878245   G   A      A  missense_variant  MODERATE   \n",
      "24     12   40657700   rs7308720   C   G      G  missense_variant  MODERATE   \n",
      "25     16   69745145   rs1800566   C   T      T  missense_variant  MODERATE   \n",
      "26     17   42987524   rs9916491   A   G      G  missense_variant  MODERATE   \n",
      "27     17   44067382   rs2258689   T   C      C  missense_variant  MODERATE   \n",
      "28     19    7141775   rs2229431   C   T      T  missense_variant  MODERATE   \n",
      "29     19   45411941    rs429358   T   C      C  missense_variant  MODERATE   \n",
      "30     19   45412079      rs7412   C   T      T  missense_variant  MODERATE   \n",
      "\n",
      "      SYMBOL             Gene  ... UKCON126 UKCON128 UKCON130 UKCON143  \\\n",
      "0      MTHFR  ENSG00000177000  ...      1/1      0/0      0/1      0/1   \n",
      "1      PINK1  ENSG00000158828  ...      0/0      0/0      0/1      0/1   \n",
      "2     CAMK1G  ENSG00000008118  ...      0/0      0/0      0/1      1/1   \n",
      "3       MAP2  ENSG00000078018  ...      0/0      0/1      0/0      0/0   \n",
      "4     INPP5D  ENSG00000168918  ...      0/1      0/0      0/0      0/0   \n",
      "5      PPARG  ENSG00000132170  ...      0/1      0/0      0/0      0/0   \n",
      "6         TF  ENSG00000091513  ...      0/0      0/0      0/0      0/0   \n",
      "7         TF  ENSG00000091513  ...      0/0      0/1      1/1      0/0   \n",
      "8   PPARGC1A  ENSG00000109819  ...      1/1      0/1      0/1      0/0   \n",
      "9   PPARGC1A  ENSG00000109819  ...      0/0      0/0      0/1      0/1   \n",
      "10   C5orf45  ENSG00000161010  ...      1/1      1/1      0/0      0/0   \n",
      "11       HFE  ENSG00000010704  ...      0/0      0/1      0/0      0/0   \n",
      "12     GSTA4  ENSG00000170899  ...      0/0      0/0      0/0      0/0   \n",
      "13     PARK2  ENSG00000185345  ...      0/0      0/0      0/0      0/0   \n",
      "14     PARK2  ENSG00000185345  ...      0/1      0/0      0/0      0/0   \n",
      "15     EPHA1  ENSG00000146904  ...      0/0      0/0      0/0      0/0   \n",
      "16      DPP6             1804  ...      0/0      0/0      0/0      1/1   \n",
      "17      DPP6  ENSG00000130226  ...      0/0      0/1      0/0      0/0   \n",
      "18      SETX  ENSG00000107290  ...      0/0      0/1      0/0      0/0   \n",
      "19      SETX  ENSG00000107290  ...      0/0      0/1      0/1      0/0   \n",
      "20      SETX  ENSG00000107290  ...      0/0      0/1      0/1      0/0   \n",
      "21      BDNF  ENSG00000176697  ...      0/1      0/1      0/0      0/1   \n",
      "22    LRRC23  ENSG00000010626  ...      0/0      0/0      1/1      0/0   \n",
      "23     LRRK2  ENSG00000188906  ...      0/0      0/0      0/1      0/1   \n",
      "24     LRRK2  ENSG00000188906  ...      0/0      0/0      0/0      0/0   \n",
      "25      NQO1  ENSG00000181019  ...      0/0      0/0      0/0      0/1   \n",
      "26      GFAP  ENSG00000131095  ...      0/0      0/0      0/0      0/0   \n",
      "27      MAPT  ENSG00000186868  ...      0/1      0/0      0/0      0/1   \n",
      "28      INSR  ENSG00000171105  ...      0/0      0/0      0/0      0/0   \n",
      "29      APOE  ENSG00000130203  ...      0/0      0/0      0/0      0/0   \n",
      "30      APOE  ENSG00000130203  ...      0/0      0/0      0/0      0/0   \n",
      "\n",
      "   UKCON144 UKCON135 UKCON136  UKCON137  UKCON138  UKCON139  \n",
      "0       0/1      0/1      NaN       NaN       NaN       NaN  \n",
      "1       1/1      0/1      NaN       NaN       NaN       NaN  \n",
      "2       0/1      0/1      NaN       NaN       NaN       NaN  \n",
      "3       0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "4       0/1      0/0      NaN       NaN       NaN       NaN  \n",
      "5       0/0      0/1      NaN       NaN       NaN       NaN  \n",
      "6       0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "7       0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "8       0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "9       0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "10      0/0      0/1      NaN       NaN       NaN       NaN  \n",
      "11      0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "12      0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "13      0/1      0/0      NaN       NaN       NaN       NaN  \n",
      "14      0/1      0/1      NaN       NaN       NaN       NaN  \n",
      "15      0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "16      0/0      0/1      NaN       NaN       NaN       NaN  \n",
      "17      0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "18      0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "19      0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "20      0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "21      0/0      1/1      NaN       NaN       NaN       NaN  \n",
      "22      0/0      0/1      NaN       NaN       NaN       NaN  \n",
      "23      0/1      0/1      NaN       NaN       NaN       NaN  \n",
      "24      0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "25      0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "26      0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "27      0/1      0/1      NaN       NaN       NaN       NaN  \n",
      "28      0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "29      0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "30      0/0      0/0      NaN       NaN       NaN       NaN  \n",
      "\n",
      "[31 rows x 1679 columns]\n",
      "Index(['Gene', 'Feature_type', 'Feature', 'BIOTYPE', 'EXON', 'INTRON', 'HGVSc',\n",
      "       'HGVSp', 'cDNA_position', 'CDS_position', 'Protein_position',\n",
      "       'Amino_acids', 'Codons', 'Existing_variation', 'DISTANCE', 'STRAND',\n",
      "       'FLAGS', 'VARIANT_CLASS', 'SYMBOL_SOURCE', 'HGNC_ID', 'CANONICAL',\n",
      "       'MANE_SELECT', 'MANE_PLUS_CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP',\n",
      "       'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT_ISOFORM', 'REFSEQ_MATCH',\n",
      "       'SOURCE', 'REFSEQ_OFFSET', 'GIVEN_REF', 'USED_REF', 'BAM_EDIT',\n",
      "       'GENE_PHENO'],\n",
      "      dtype='object')\n",
      "    CHROM        POS          ID REF ALT Allele       Consequence    IMPACT  \\\n",
      "0       1   11854476   rs1801131   A   C      C  missense_variant  MODERATE   \n",
      "1       1   20977000   rs1043424   A   C      C  missense_variant  MODERATE   \n",
      "2       1  209782343  rs11119314   T   C      C  missense_variant  MODERATE   \n",
      "3       2  210558162    rs741006   G   A      A  missense_variant  MODERATE   \n",
      "4       2  234113301      rs9247   C   T      T  missense_variant  MODERATE   \n",
      "5       3   12393125   rs1801282   C   G      G  missense_variant  MODERATE   \n",
      "6       3  133475722   rs1799852   G   A      A  missense_variant  MODERATE   \n",
      "7       3  133494354   rs1049296   C   T      T  missense_variant  MODERATE   \n",
      "8       4   23815662   rs8192678   C   T      T  missense_variant  MODERATE   \n",
      "9       4   23815681  rs17574213   C   T      T  missense_variant  MODERATE   \n",
      "10      5  179264731     rs10277   G   A      A  missense_variant  MODERATE   \n",
      "11      6   26093141   rs1800562   C   T      T  missense_variant  MODERATE   \n",
      "12      6   52849325   rs1802061   G   A      A  missense_variant  MODERATE   \n",
      "13      6  161781225   rs1801334   G   A      A  missense_variant  MODERATE   \n",
      "14      6  161807855   rs1801582   C   G      G  missense_variant  MODERATE   \n",
      "15      7  143095153  rs11768549   G   A      A  missense_variant  MODERATE   \n",
      "16      7  154587793  rs10236495   A   G      G  missense_variant  MODERATE   \n",
      "17      7  154672646   rs3817522   A   G      G  missense_variant  MODERATE   \n",
      "18      9  135139826   rs3739927   T   C      C  missense_variant  MODERATE   \n",
      "19      9  135172412   rs2296869   T   C      C  missense_variant  MODERATE   \n",
      "20      9  135173685   rs2296871   A   G      G  missense_variant  MODERATE   \n",
      "21     11   27679916      rs6265   C   T      T  missense_variant  MODERATE   \n",
      "22     12    7023123    rs710415   A   G      G  missense_variant  MODERATE   \n",
      "23     12   40631791  rs10878245   G   A      A  missense_variant  MODERATE   \n",
      "24     12   40657700   rs7308720   C   G      G  missense_variant  MODERATE   \n",
      "25     16   69745145   rs1800566   C   T      T  missense_variant  MODERATE   \n",
      "26     17   42987524   rs9916491   A   G      G  missense_variant  MODERATE   \n",
      "27     17   44067382   rs2258689   T   C      C  missense_variant  MODERATE   \n",
      "28     19    7141775   rs2229431   C   T      T  missense_variant  MODERATE   \n",
      "29     19   45411941    rs429358   T   C      C  missense_variant  MODERATE   \n",
      "30     19   45412079      rs7412   C   T      T  missense_variant  MODERATE   \n",
      "\n",
      "      SYMBOL                              SIFT  ... UKCON126 UKCON128  \\\n",
      "0      MTHFR                 deleterious(0.01)  ...      1/1      0/0   \n",
      "1      PINK1                   tolerated(0.42)  ...      0/0      0/0   \n",
      "2     CAMK1G  deleterious_low_confidence(0.01)  ...      0/0      0/0   \n",
      "3       MAP2       tolerated_low_confidence(1)  ...      0/0      0/1   \n",
      "4     INPP5D    tolerated_low_confidence(0.06)  ...      0/1      0/0   \n",
      "5      PPARG    tolerated_low_confidence(0.05)  ...      0/1      0/0   \n",
      "6         TF                   tolerated(0.11)  ...      0/0      0/0   \n",
      "7         TF                   tolerated(0.13)  ...      0/0      0/1   \n",
      "8   PPARGC1A    tolerated_low_confidence(0.98)  ...      1/1      0/1   \n",
      "9   PPARGC1A                   tolerated(0.32)  ...      0/0      0/0   \n",
      "10   C5orf45                 deleterious(0.05)  ...      1/1      1/1   \n",
      "11       HFE                 deleterious(0.01)  ...      0/0      0/1   \n",
      "12     GSTA4                    deleterious(0)  ...      0/0      0/0   \n",
      "13     PARK2                    deleterious(0)  ...      0/0      0/0   \n",
      "14     PARK2                   tolerated(0.48)  ...      0/1      0/0   \n",
      "15     EPHA1                   tolerated(0.25)  ...      0/0      0/0   \n",
      "16      DPP6                                 .  ...      0/0      0/0   \n",
      "17      DPP6                    deleterious(0)  ...      0/0      0/1   \n",
      "18      SETX    tolerated_low_confidence(0.66)  ...      0/0      0/1   \n",
      "19      SETX                   tolerated(0.08)  ...      0/0      0/1   \n",
      "20      SETX                   tolerated(0.11)  ...      0/0      0/1   \n",
      "21      BDNF                    tolerated(0.2)  ...      0/1      0/1   \n",
      "22    LRRC23    tolerated_low_confidence(0.49)  ...      0/0      0/0   \n",
      "23     LRRK2                 deleterious(0.03)  ...      0/0      0/0   \n",
      "24     LRRK2                 deleterious(0.01)  ...      0/0      0/0   \n",
      "25      NQO1                    deleterious(0)  ...      0/0      0/0   \n",
      "26      GFAP                 deleterious(0.01)  ...      0/0      0/0   \n",
      "27      MAPT                      tolerated(1)  ...      0/1      0/0   \n",
      "28      INSR                   tolerated(0.08)  ...      0/0      0/0   \n",
      "29      APOE                      tolerated(1)  ...      0/0      0/0   \n",
      "30      APOE                    deleterious(0)  ...      0/0      0/0   \n",
      "\n",
      "   UKCON130 UKCON143 UKCON144 UKCON135 UKCON136 UKCON137 UKCON138 UKCON139  \n",
      "0       0/1      0/1      0/1      0/1      NaN      NaN      NaN      NaN  \n",
      "1       0/1      0/1      1/1      0/1      NaN      NaN      NaN      NaN  \n",
      "2       0/1      1/1      0/1      0/1      NaN      NaN      NaN      NaN  \n",
      "3       0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "4       0/0      0/0      0/1      0/0      NaN      NaN      NaN      NaN  \n",
      "5       0/0      0/0      0/0      0/1      NaN      NaN      NaN      NaN  \n",
      "6       0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "7       1/1      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "8       0/1      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "9       0/1      0/1      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "10      0/0      0/0      0/0      0/1      NaN      NaN      NaN      NaN  \n",
      "11      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "12      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "13      0/0      0/0      0/1      0/0      NaN      NaN      NaN      NaN  \n",
      "14      0/0      0/0      0/1      0/1      NaN      NaN      NaN      NaN  \n",
      "15      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "16      0/0      1/1      0/0      0/1      NaN      NaN      NaN      NaN  \n",
      "17      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "18      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "19      0/1      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "20      0/1      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "21      0/0      0/1      0/0      1/1      NaN      NaN      NaN      NaN  \n",
      "22      1/1      0/0      0/0      0/1      NaN      NaN      NaN      NaN  \n",
      "23      0/1      0/1      0/1      0/1      NaN      NaN      NaN      NaN  \n",
      "24      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "25      0/0      0/1      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "26      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "27      0/0      0/1      0/1      0/1      NaN      NaN      NaN      NaN  \n",
      "28      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "29      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "30      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "\n",
      "[31 rows x 1641 columns]\n",
      "    CHROM        POS          ID REF ALT Allele       Consequence    IMPACT  \\\n",
      "0       1   11854476   rs1801131   A   C      C  missense_variant  MODERATE   \n",
      "1       1   20977000   rs1043424   A   C      C  missense_variant  MODERATE   \n",
      "2       1  209782343  rs11119314   T   C      C  missense_variant  MODERATE   \n",
      "3       2  210558162    rs741006   G   A      A  missense_variant  MODERATE   \n",
      "4       2  234113301      rs9247   C   T      T  missense_variant  MODERATE   \n",
      "5       3   12393125   rs1801282   C   G      G  missense_variant  MODERATE   \n",
      "6       3  133475722   rs1799852   G   A      A  missense_variant  MODERATE   \n",
      "7       3  133494354   rs1049296   C   T      T  missense_variant  MODERATE   \n",
      "8       4   23815662   rs8192678   C   T      T  missense_variant  MODERATE   \n",
      "9       4   23815681  rs17574213   C   T      T  missense_variant  MODERATE   \n",
      "10      5  179264731     rs10277   G   A      A  missense_variant  MODERATE   \n",
      "11      6   26093141   rs1800562   C   T      T  missense_variant  MODERATE   \n",
      "12      6   52849325   rs1802061   G   A      A  missense_variant  MODERATE   \n",
      "13      6  161781225   rs1801334   G   A      A  missense_variant  MODERATE   \n",
      "14      6  161807855   rs1801582   C   G      G  missense_variant  MODERATE   \n",
      "15      7  143095153  rs11768549   G   A      A  missense_variant  MODERATE   \n",
      "16      7  154587793  rs10236495   A   G      G  missense_variant  MODERATE   \n",
      "17      7  154672646   rs3817522   A   G      G  missense_variant  MODERATE   \n",
      "18      9  135139826   rs3739927   T   C      C  missense_variant  MODERATE   \n",
      "19      9  135172412   rs2296869   T   C      C  missense_variant  MODERATE   \n",
      "20      9  135173685   rs2296871   A   G      G  missense_variant  MODERATE   \n",
      "21     11   27679916      rs6265   C   T      T  missense_variant  MODERATE   \n",
      "22     12    7023123    rs710415   A   G      G  missense_variant  MODERATE   \n",
      "23     12   40631791  rs10878245   G   A      A  missense_variant  MODERATE   \n",
      "24     12   40657700   rs7308720   C   G      G  missense_variant  MODERATE   \n",
      "25     16   69745145   rs1800566   C   T      T  missense_variant  MODERATE   \n",
      "26     17   42987524   rs9916491   A   G      G  missense_variant  MODERATE   \n",
      "27     17   44067382   rs2258689   T   C      C  missense_variant  MODERATE   \n",
      "28     19    7141775   rs2229431   C   T      T  missense_variant  MODERATE   \n",
      "29     19   45411941    rs429358   T   C      C  missense_variant  MODERATE   \n",
      "30     19   45412079      rs7412   C   T      T  missense_variant  MODERATE   \n",
      "\n",
      "      SYMBOL                              SIFT  ... UKCON126 UKCON128  \\\n",
      "0      MTHFR                 deleterious(0.01)  ...      1/1      0/0   \n",
      "1      PINK1                   tolerated(0.42)  ...      0/0      0/0   \n",
      "2     CAMK1G  deleterious_low_confidence(0.01)  ...      0/0      0/0   \n",
      "3       MAP2       tolerated_low_confidence(1)  ...      0/0      0/1   \n",
      "4     INPP5D    tolerated_low_confidence(0.06)  ...      0/1      0/0   \n",
      "5      PPARG    tolerated_low_confidence(0.05)  ...      0/1      0/0   \n",
      "6         TF                   tolerated(0.11)  ...      0/0      0/0   \n",
      "7         TF                   tolerated(0.13)  ...      0/0      0/1   \n",
      "8   PPARGC1A    tolerated_low_confidence(0.98)  ...      1/1      0/1   \n",
      "9   PPARGC1A                   tolerated(0.32)  ...      0/0      0/0   \n",
      "10   C5orf45                 deleterious(0.05)  ...      1/1      1/1   \n",
      "11       HFE                 deleterious(0.01)  ...      0/0      0/1   \n",
      "12     GSTA4                    deleterious(0)  ...      0/0      0/0   \n",
      "13     PARK2                    deleterious(0)  ...      0/0      0/0   \n",
      "14     PARK2                   tolerated(0.48)  ...      0/1      0/0   \n",
      "15     EPHA1                   tolerated(0.25)  ...      0/0      0/0   \n",
      "16      DPP6                                 .  ...      0/0      0/0   \n",
      "17      DPP6                    deleterious(0)  ...      0/0      0/1   \n",
      "18      SETX    tolerated_low_confidence(0.66)  ...      0/0      0/1   \n",
      "19      SETX                   tolerated(0.08)  ...      0/0      0/1   \n",
      "20      SETX                   tolerated(0.11)  ...      0/0      0/1   \n",
      "21      BDNF                    tolerated(0.2)  ...      0/1      0/1   \n",
      "22    LRRC23    tolerated_low_confidence(0.49)  ...      0/0      0/0   \n",
      "23     LRRK2                 deleterious(0.03)  ...      0/0      0/0   \n",
      "24     LRRK2                 deleterious(0.01)  ...      0/0      0/0   \n",
      "25      NQO1                    deleterious(0)  ...      0/0      0/0   \n",
      "26      GFAP                 deleterious(0.01)  ...      0/0      0/0   \n",
      "27      MAPT                      tolerated(1)  ...      0/1      0/0   \n",
      "28      INSR                   tolerated(0.08)  ...      0/0      0/0   \n",
      "29      APOE                      tolerated(1)  ...      0/0      0/0   \n",
      "30      APOE                    deleterious(0)  ...      0/0      0/0   \n",
      "\n",
      "   UKCON130 UKCON143 UKCON144 UKCON135 UKCON136 UKCON137 UKCON138 UKCON139  \n",
      "0       0/1      0/1      0/1      0/1      NaN      NaN      NaN      NaN  \n",
      "1       0/1      0/1      1/1      0/1      NaN      NaN      NaN      NaN  \n",
      "2       0/1      1/1      0/1      0/1      NaN      NaN      NaN      NaN  \n",
      "3       0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "4       0/0      0/0      0/1      0/0      NaN      NaN      NaN      NaN  \n",
      "5       0/0      0/0      0/0      0/1      NaN      NaN      NaN      NaN  \n",
      "6       0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "7       1/1      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "8       0/1      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "9       0/1      0/1      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "10      0/0      0/0      0/0      0/1      NaN      NaN      NaN      NaN  \n",
      "11      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "12      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "13      0/0      0/0      0/1      0/0      NaN      NaN      NaN      NaN  \n",
      "14      0/0      0/0      0/1      0/1      NaN      NaN      NaN      NaN  \n",
      "15      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "16      0/0      1/1      0/0      0/1      NaN      NaN      NaN      NaN  \n",
      "17      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "18      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "19      0/1      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "20      0/1      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "21      0/0      0/1      0/0      1/1      NaN      NaN      NaN      NaN  \n",
      "22      1/1      0/0      0/0      0/1      NaN      NaN      NaN      NaN  \n",
      "23      0/1      0/1      0/1      0/1      NaN      NaN      NaN      NaN  \n",
      "24      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "25      0/0      0/1      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "26      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "27      0/0      0/1      0/1      0/1      NaN      NaN      NaN      NaN  \n",
      "28      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "29      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "30      0/0      0/0      0/0      0/0      NaN      NaN      NaN      NaN  \n",
      "\n",
      "[31 rows x 1610 columns]\n",
      "    CHROM        POS          ID REF ALT       Consequence    SYMBOL  \\\n",
      "0       1   11854476   rs1801131   A   C  missense_variant     MTHFR   \n",
      "1       1   20977000   rs1043424   A   C  missense_variant     PINK1   \n",
      "2       1  209782343  rs11119314   T   C  missense_variant    CAMK1G   \n",
      "3       2  210558162    rs741006   G   A  missense_variant      MAP2   \n",
      "4       2  234113301      rs9247   C   T  missense_variant    INPP5D   \n",
      "5       3   12393125   rs1801282   C   G  missense_variant     PPARG   \n",
      "6       3  133475722   rs1799852   G   A  missense_variant        TF   \n",
      "7       3  133494354   rs1049296   C   T  missense_variant        TF   \n",
      "8       4   23815662   rs8192678   C   T  missense_variant  PPARGC1A   \n",
      "9       4   23815681  rs17574213   C   T  missense_variant  PPARGC1A   \n",
      "10      5  179264731     rs10277   G   A  missense_variant   C5orf45   \n",
      "11      6   26093141   rs1800562   C   T  missense_variant       HFE   \n",
      "12      6   52849325   rs1802061   G   A  missense_variant     GSTA4   \n",
      "13      6  161781225   rs1801334   G   A  missense_variant     PARK2   \n",
      "14      6  161807855   rs1801582   C   G  missense_variant     PARK2   \n",
      "15      7  143095153  rs11768549   G   A  missense_variant     EPHA1   \n",
      "16      7  154587793  rs10236495   A   G  missense_variant      DPP6   \n",
      "17      7  154672646   rs3817522   A   G  missense_variant      DPP6   \n",
      "18      9  135139826   rs3739927   T   C  missense_variant      SETX   \n",
      "19      9  135172412   rs2296869   T   C  missense_variant      SETX   \n",
      "20      9  135173685   rs2296871   A   G  missense_variant      SETX   \n",
      "21     11   27679916      rs6265   C   T  missense_variant      BDNF   \n",
      "22     12    7023123    rs710415   A   G  missense_variant    LRRC23   \n",
      "23     12   40631791  rs10878245   G   A  missense_variant     LRRK2   \n",
      "24     12   40657700   rs7308720   C   G  missense_variant     LRRK2   \n",
      "25     16   69745145   rs1800566   C   T  missense_variant      NQO1   \n",
      "26     17   42987524   rs9916491   A   G  missense_variant      GFAP   \n",
      "27     17   44067382   rs2258689   T   C  missense_variant      MAPT   \n",
      "28     19    7141775   rs2229431   C   T  missense_variant      INSR   \n",
      "29     19   45411941    rs429358   T   C  missense_variant      APOE   \n",
      "30     19   45412079      rs7412   C   T  missense_variant      APOE   \n",
      "\n",
      "                                SIFT                  PolyPhen EURCON115  ...  \\\n",
      "0                  deleterious(0.01)  possibly_damaging(0.654)       1/1  ...   \n",
      "1                    tolerated(0.42)             benign(0.006)       0/0  ...   \n",
      "2   deleterious_low_confidence(0.01)                unknown(0)       1/1  ...   \n",
      "3        tolerated_low_confidence(1)                 benign(0)       0/0  ...   \n",
      "4     tolerated_low_confidence(0.06)  possibly_damaging(0.814)       0/1  ...   \n",
      "5     tolerated_low_confidence(0.05)                 benign(0)       0/0  ...   \n",
      "6                    tolerated(0.11)             benign(0.096)       0/0  ...   \n",
      "7                    tolerated(0.13)             benign(0.018)       0/1  ...   \n",
      "8     tolerated_low_confidence(0.98)             benign(0.003)       0/0  ...   \n",
      "9                    tolerated(0.32)             benign(0.019)       0/0  ...   \n",
      "10                 deleterious(0.05)             benign(0.001)       0/1  ...   \n",
      "11                 deleterious(0.01)  probably_damaging(0.967)       0/0  ...   \n",
      "12                    deleterious(0)  possibly_damaging(0.803)       0/0  ...   \n",
      "13                    deleterious(0)  probably_damaging(0.977)       0/0  ...   \n",
      "14                   tolerated(0.48)                 benign(0)       0/1  ...   \n",
      "15                   tolerated(0.25)             benign(0.044)       0/0  ...   \n",
      "16                                 .                         .       0/1  ...   \n",
      "17                    deleterious(0)   possibly_damaging(0.83)       0/1  ...   \n",
      "18    tolerated_low_confidence(0.66)                 benign(0)       0/0  ...   \n",
      "19                   tolerated(0.08)             benign(0.212)       0/0  ...   \n",
      "20                   tolerated(0.11)             benign(0.012)       0/0  ...   \n",
      "21                    tolerated(0.2)  probably_damaging(0.991)       0/0  ...   \n",
      "22    tolerated_low_confidence(0.49)                 benign(0)       0/0  ...   \n",
      "23                 deleterious(0.03)  possibly_damaging(0.704)       0/1  ...   \n",
      "24                 deleterious(0.01)  probably_damaging(0.972)       0/0  ...   \n",
      "25                    deleterious(0)  possibly_damaging(0.849)       0/0  ...   \n",
      "26                 deleterious(0.01)                 benign(0)       1/1  ...   \n",
      "27                      tolerated(1)                 benign(0)       1/1  ...   \n",
      "28                   tolerated(0.08)             benign(0.111)       0/0  ...   \n",
      "29                      tolerated(1)             benign(0.001)       0/0  ...   \n",
      "30                    deleterious(0)      probably_damaging(1)       0/0  ...   \n",
      "\n",
      "   UKCON126 UKCON128 UKCON130 UKCON143 UKCON144 UKCON135 UKCON136 UKCON137  \\\n",
      "0       1/1      0/0      0/1      0/1      0/1      0/1      NaN      NaN   \n",
      "1       0/0      0/0      0/1      0/1      1/1      0/1      NaN      NaN   \n",
      "2       0/0      0/0      0/1      1/1      0/1      0/1      NaN      NaN   \n",
      "3       0/0      0/1      0/0      0/0      0/0      0/0      NaN      NaN   \n",
      "4       0/1      0/0      0/0      0/0      0/1      0/0      NaN      NaN   \n",
      "5       0/1      0/0      0/0      0/0      0/0      0/1      NaN      NaN   \n",
      "6       0/0      0/0      0/0      0/0      0/0      0/0      NaN      NaN   \n",
      "7       0/0      0/1      1/1      0/0      0/0      0/0      NaN      NaN   \n",
      "8       1/1      0/1      0/1      0/0      0/0      0/0      NaN      NaN   \n",
      "9       0/0      0/0      0/1      0/1      0/0      0/0      NaN      NaN   \n",
      "10      1/1      1/1      0/0      0/0      0/0      0/1      NaN      NaN   \n",
      "11      0/0      0/1      0/0      0/0      0/0      0/0      NaN      NaN   \n",
      "12      0/0      0/0      0/0      0/0      0/0      0/0      NaN      NaN   \n",
      "13      0/0      0/0      0/0      0/0      0/1      0/0      NaN      NaN   \n",
      "14      0/1      0/0      0/0      0/0      0/1      0/1      NaN      NaN   \n",
      "15      0/0      0/0      0/0      0/0      0/0      0/0      NaN      NaN   \n",
      "16      0/0      0/0      0/0      1/1      0/0      0/1      NaN      NaN   \n",
      "17      0/0      0/1      0/0      0/0      0/0      0/0      NaN      NaN   \n",
      "18      0/0      0/1      0/0      0/0      0/0      0/0      NaN      NaN   \n",
      "19      0/0      0/1      0/1      0/0      0/0      0/0      NaN      NaN   \n",
      "20      0/0      0/1      0/1      0/0      0/0      0/0      NaN      NaN   \n",
      "21      0/1      0/1      0/0      0/1      0/0      1/1      NaN      NaN   \n",
      "22      0/0      0/0      1/1      0/0      0/0      0/1      NaN      NaN   \n",
      "23      0/0      0/0      0/1      0/1      0/1      0/1      NaN      NaN   \n",
      "24      0/0      0/0      0/0      0/0      0/0      0/0      NaN      NaN   \n",
      "25      0/0      0/0      0/0      0/1      0/0      0/0      NaN      NaN   \n",
      "26      0/0      0/0      0/0      0/0      0/0      0/0      NaN      NaN   \n",
      "27      0/1      0/0      0/0      0/1      0/1      0/1      NaN      NaN   \n",
      "28      0/0      0/0      0/0      0/0      0/0      0/0      NaN      NaN   \n",
      "29      0/0      0/0      0/0      0/0      0/0      0/0      NaN      NaN   \n",
      "30      0/0      0/0      0/0      0/0      0/0      0/0      NaN      NaN   \n",
      "\n",
      "   UKCON138 UKCON139  \n",
      "0       NaN      NaN  \n",
      "1       NaN      NaN  \n",
      "2       NaN      NaN  \n",
      "3       NaN      NaN  \n",
      "4       NaN      NaN  \n",
      "5       NaN      NaN  \n",
      "6       NaN      NaN  \n",
      "7       NaN      NaN  \n",
      "8       NaN      NaN  \n",
      "9       NaN      NaN  \n",
      "10      NaN      NaN  \n",
      "11      NaN      NaN  \n",
      "12      NaN      NaN  \n",
      "13      NaN      NaN  \n",
      "14      NaN      NaN  \n",
      "15      NaN      NaN  \n",
      "16      NaN      NaN  \n",
      "17      NaN      NaN  \n",
      "18      NaN      NaN  \n",
      "19      NaN      NaN  \n",
      "20      NaN      NaN  \n",
      "21      NaN      NaN  \n",
      "22      NaN      NaN  \n",
      "23      NaN      NaN  \n",
      "24      NaN      NaN  \n",
      "25      NaN      NaN  \n",
      "26      NaN      NaN  \n",
      "27      NaN      NaN  \n",
      "28      NaN      NaN  \n",
      "29      NaN      NaN  \n",
      "30      NaN      NaN  \n",
      "\n",
      "[31 rows x 1608 columns]\n",
      "missense\n",
      "(139, 1599)\n",
      "Creating samples graphs...\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m~/anaconda3/envs/graphgym/lib/python3.7/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m__setattr__\u001b[0;34m(self, name, value)\u001b[0m\n\u001b[1;32m   5476\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 5477\u001b[0;31m             \u001b[0mobject\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__getattribute__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   5478\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mobject\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__setattr__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/graphgym/lib/python3.7/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36mname\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    535\u001b[0m         \"\"\"\n\u001b[0;32m--> 536\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_name\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    537\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/graphgym/lib/python3.7/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m   5460\u001b[0m         ):\n\u001b[0;32m-> 5461\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mobject\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__getattribute__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   5462\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'Series' object has no attribute '_name'",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-5-dc511a1c9566>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     15\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     16\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 17\u001b[0;31m     \u001b[0mresult_nodes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcreate_datasets\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcreate_nx_datasets\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmain\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindir\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdataset\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtarget\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdisease\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnetwork\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'missense'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     18\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Coding: number of missense variants per node'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     19\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/CODE/graph_genetics/create_datasets/create_nx_datasets.py\u001b[0m in \u001b[0;36mmain\u001b[0;34m(indir, dataset, target, disease, network, mode, number)\u001b[0m\n\u001b[1;32m    330\u001b[0m         \u001b[0medges_attr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    331\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 332\u001b[0;31m         \u001b[0mresult_graphs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcreate_samples_graphs\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'missense'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnodes_attr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mppi_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdiagnosis\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtarget\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    333\u001b[0m         \u001b[0mresult_graphs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdelete_small_components\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresult_graphs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m4\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    334\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/CODE/graph_genetics/create_datasets/create_nx_datasets.py\u001b[0m in \u001b[0;36mcreate_samples_graphs\u001b[0;34m(mode, nodes_matrix, edges_matrix, original_graph, diagnosis, target)\u001b[0m\n\u001b[1;32m    217\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mmode\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'missense'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    218\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0mn\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mnodes\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 219\u001b[0;31m                 \u001b[0msample_graph\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnodes\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mn\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'node_feature'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtorch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtensor\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mnodes_matrix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mn\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0msample\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# missense\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    220\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    221\u001b[0m         \u001b[0mgraphs_list\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msample_graph\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/graphgym/lib/python3.7/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m    893\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    894\u001b[0m             \u001b[0mmaybe_callable\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_if_callable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 895\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_axis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmaybe_callable\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    896\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    897\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_is_scalar_access\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mTuple\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/graphgym/lib/python3.7/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_getitem_axis\u001b[0;34m(self, key, axis)\u001b[0m\n\u001b[1;32m   1122\u001b[0m         \u001b[0;31m# fall thru to straight lookup\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1123\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_validate_key\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1124\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_label\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1125\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1126\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_get_slice_axis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mslice_obj\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mslice\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/graphgym/lib/python3.7/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_get_label\u001b[0;34m(self, label, axis)\u001b[0m\n\u001b[1;32m   1071\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_get_label\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlabel\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1072\u001b[0m         \u001b[0;31m# GH#5667 this will fail if the label is not present in the axis.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1073\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mxs\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlabel\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1074\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1075\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_handle_lowerdim_multi_index_axis0\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtup\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mTuple\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/graphgym/lib/python3.7/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36mxs\u001b[0;34m(self, key, axis, level, drop_level)\u001b[0m\n\u001b[1;32m   3763\u001b[0m                 \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3764\u001b[0m                 \u001b[0mname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3765\u001b[0;31m                 \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mnew_values\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3766\u001b[0m             )\n\u001b[1;32m   3767\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0mis_scalar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/graphgym/lib/python3.7/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, data, index, dtype, name, copy, fastpath)\u001b[0m\n\u001b[1;32m    367\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    368\u001b[0m         \u001b[0mgeneric\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mNDFrame\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 369\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    370\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_set_axis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfastpath\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    371\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/graphgym/lib/python3.7/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m__setattr__\u001b[0;34m(self, name, value)\u001b[0m\n\u001b[1;32m   5475\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5476\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 5477\u001b[0;31m             \u001b[0mobject\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__getattribute__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   5478\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mobject\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__setattr__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5479\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mAttributeError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "dataset = 'LOAD'\n",
    "target  = 'LOAD'\n",
    "diseases = ['ND']\n",
    "network = 'original'\n",
    "\n",
    "for disease in diseases:\n",
    "\n",
    "    indir = 'data'\n",
    "    outdir = f'data/graph_datasets/{target}'\n",
    "    print('Input directory:', indir)\n",
    "    print('Output directory:', outdir)\n",
    "    print()\n",
    "\n",
    "    start_time = datetime.datetime.now()\n",
    "    print()\n",
    "\n",
    "    result_nodes = create_datasets.create_nx_datasets.main(indir, dataset, target, disease, network, 'missense', None)\n",
    "    print('Coding: number of missense variants per node')\n",
    "\n",
    "    outfile = f'{outdir}/{disease}_PPI_missense.pkl'\n",
    "    print('Resulting dataset saved at:', outfile)\n",
    "    print()\n",
    "\n",
    "    with open(outfile, 'wb') as f:\n",
    "        pickle.dump(result_nodes, f)\n",
    "\n",
    "    result_nodes_time = datetime.datetime.now()\n",
    "    print('Processing time:', result_nodes_time - start_time)\n",
    "    print('\\n\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**5. Graph classification with GNNs**\n",
    "\n",
    "We then evaluated and tested different GNNs in the framework called [GraphGym](https://github.com/snap-stanford/GraphGym) (You *et al.*, 2020).\n",
    "\n",
    "Configuration and grid files employed are in the subdirectory [graphgym_files](graphgym_files).\n",
    "\n",
    "Summarized results obtained by GraphGym in LOAD dataset are [here](results/GNNs_LOAD/2022_02_LOAD.csv)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**6. Baseline model**\n",
    "\n",
    "Using only APOE gene as input."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Class distribution:\n",
      "1.0    1014\n",
      "0.0     585\n",
      "Name: y, dtype: int64\n",
      "\n",
      "Confusion matrix:\n",
      " [[  0  59]\n",
      " [  0 101]]\n",
      "\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "         0.0       0.00      0.00      0.00        59\n",
      "         1.0       0.63      1.00      0.77       101\n",
      "\n",
      "    accuracy                           0.63       160\n",
      "   macro avg       0.32      0.50      0.39       160\n",
      "weighted avg       0.40      0.63      0.49       160\n",
      "\n",
      "Baseline model LOAD, AUC ROC: 0.5402752139620742\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/laura/anaconda3/envs/graphgym/lib/python3.7/site-packages/sklearn/metrics/_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/home/laura/anaconda3/envs/graphgym/lib/python3.7/site-packages/sklearn/metrics/_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/home/laura/anaconda3/envs/graphgym/lib/python3.7/site-packages/sklearn/metrics/_classification.py:1245: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    }
   ],
   "source": [
    "target = 'LOAD'\n",
    "\n",
    "infile = f'data/table_datasets/ND_PPI_missense_with_biomarkers_LOAD.csv'\n",
    "data = pd.read_csv(infile, index_col = 0)\n",
    "data_wclass = create_class_LOAD(data)\n",
    "\n",
    "x = data_wclass.drop(columns=['y'])\n",
    "x = x['APOE']\n",
    "\n",
    "y = data_wclass['y']\n",
    "x.index = x.index.str.upper()\n",
    "\n",
    "f = open(f'data/splits/split_{target}.pkl', 'rb')\n",
    "split_load = pickle.load(f)\n",
    "f.close()\n",
    "\n",
    "auc_load = baseline_model(split_load, x, y)\n",
    "print('Baseline model LOAD, AUC ROC:', auc_load)"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "5e0466fb38f8b8207bee76f1d5f01e5f82e3be8917a80d900a987709dfd81e7a"
  },
  "kernelspec": {
   "display_name": "Python 3.7.10 ('graphgym')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
