{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Main methodology results\n",
    "\n",
    "*Disclaimer*: please note that many parts of this code require the preprocessed data from ADNI (both genetic and diagnostic related) as input. This data has not been uploaded to the repository for privacy reasons."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import pandas as pd\n",
    "import datetime, pickle\n",
    "import create_datasets.create_nx_datasets\n",
    "from ml_models.machine_learning_models import create_class_LOAD, baseline_model"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**1. Obtain genes of interest**\n",
    "\n",
    "Using DisGeNET to get Gene-Disease-Associations (GDAs) to Alzheimer's Disease (AD gene set) and other neurodegenerative diseases (ND). This is already obtained from [first part of the results](1_main_methodology.ipynb)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**2. Obtain biological networks**\n",
    "\n",
    "Using genes of interst obtained from DisGeNET, obtain PPI between them from STRING. This is already obtained from [first part of the results](1_main_methodology.ipynb)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**3. Data preprocessing**\n",
    "\n",
    "Please refer to `data_preprocessing` subdirectory for this part.\n",
    "1. [make_BED_files.R](data_preprocessing/make_BED_files.R) creates BED files with the genomic coordinates of the genes of interest. This is already obtained from [first part of the results](1_main_methodology.ipynb).\n",
    "2. [extract_and_annotate_missense_LOAD.sh](data_preprocessing/extract_and_annotate_missense_LOAD.sh) is the script for obtaining missense variants from the VCF files."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**4. Create graph datasets**\n",
    "\n",
    "Create graph datasets (one graph representing each patient) for different targets with ADNI dataset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Input directory: data\n",
      "Output directory: data/graph_datasets/LOAD\n",
      "\n",
      "\n",
      "data/ND_STRING_PPI_edgelist.txt\n",
      "Network used: ND original\n",
      "# nodes = 139\n",
      "# edges = 263\n",
      "\n",
      "Dataset used: LOAD\n",
      "    CHROM        POS          ID REF ALT Allele           Consequence  \\\n",
      "0       1   11854476   rs1801131   A   C      C      missense_variant   \n",
      "1       1   20977000   rs1043424   A   C      C      missense_variant   \n",
      "2       1  209782343  rs11119314   T   C      C      missense_variant   \n",
      "3       2  210558162    rs741006   G   A      A      missense_variant   \n",
      "4       2  234113301      rs9247   C   T      T      missense_variant   \n",
      "5       3   12393125   rs1801282   C   G      G      missense_variant   \n",
      "6       3  133475722   rs1799852   G   A      A      missense_variant   \n",
      "7       3  133494354   rs1049296   C   T      T      missense_variant   \n",
      "8       4   23815662   rs8192678   C   T      T      missense_variant   \n",
      "9       4   23815681  rs17574213   C   T      T      missense_variant   \n",
      "10      5  179264731     rs10277   G   A      A      missense_variant   \n",
      "11      6   26093141   rs1800562   C   T      T      missense_variant   \n",
      "12      6   52849325   rs1802061   G   A      A      missense_variant   \n",
      "13      6  161781225   rs1801334   G   A      A      missense_variant   \n",
      "14      6  161807855   rs1801582   C   G      G      missense_variant   \n",
      "15      7  143095153  rs11768549   G   A      A      missense_variant   \n",
      "16      7  154587793  rs10236495   A   G      G      missense_variant   \n",
      "17      7  154672646   rs3817522   A   G      G      missense_variant   \n",
      "18      9  135139826   rs3739927   T   C      C      missense_variant   \n",
      "19      9  135172412   rs2296869   T   C      C      missense_variant   \n",
      "20      9  135173685   rs2296871   A   G      G      missense_variant   \n",
      "21     11   27679916      rs6265   C   T      T      missense_variant   \n",
      "22     12    7023123    rs710415   A   G      G      missense_variant   \n",
      "23     12   40631791  rs10878245   G   A      A      missense_variant   \n",
      "24     12   40657700   rs7308720   C   G      G      missense_variant   \n",
      "25     16   69745145   rs1800566   C   T      T      missense_variant   \n",
      "26     17   42987524   rs9916491   A   G      G      missense_variant   \n",
      "27     17   44067382   rs2258689   T   C      C      missense_variant   \n",
      "28     19    7141775   rs2229431   C   T      T      missense_variant   \n",
      "29     19   45411941    rs429358   T   C      C      missense_variant   \n",
      "30     19   45412079      rs7412   C   T      T      missense_variant   \n",
      "\n",
      "      IMPACT                   SYMBOL                 Gene  ... UKCON128  \\\n",
      "0   MODERATE                    MTHFR      ENSG00000177000  ...      0/1   \n",
      "1   MODERATE                    PINK1      ENSG00000158828  ...      0/0   \n",
      "2   MODERATE                   CAMK1G      ENSG00000008118  ...      0/1   \n",
      "3   MODERATE                     MAP2      ENSG00000078018  ...      0/0   \n",
      "4   MODERATE                   INPP5D      ENSG00000168918  ...      0/1   \n",
      "5   MODERATE                    PPARG      ENSG00000132170  ...      0/0   \n",
      "6   MODERATE                       TF      ENSG00000091513  ...      0/0   \n",
      "7   MODERATE                       TF      ENSG00000091513  ...      0/0   \n",
      "8   MODERATE  PPARGC1AENSG00000109819           Transcript  ...      1/1   \n",
      "9   MODERATE  PPARGC1AENSG00000109819           Transcript  ...      0/0   \n",
      "10  MODERATE                  C5orf45      ENSG00000161010  ...      0/1   \n",
      "11  MODERATE                      HFE      ENSG00000010704  ...      0/0   \n",
      "12  MODERATE                    GSTA4      ENSG00000170899  ...      0/0   \n",
      "13  MODERATE                    PARK2      ENSG00000185345  ...      0/0   \n",
      "14  MODERATE                    PARK2      ENSG00000185345  ...      0/0   \n",
      "15  MODERATE                    EPHA1      ENSG00000146904  ...      0/0   \n",
      "16  MODERATE                     DPP6                 1804  ...      1/1   \n",
      "17  MODERATE                     DPP6      ENSG00000130226  ...      0/0   \n",
      "18  MODERATE                     SETX      ENSG00000107290  ...      0/0   \n",
      "19  MODERATE                     SETX      ENSG00000107290  ...      0/1   \n",
      "20  MODERATE                     SETX      ENSG00000107290  ...      0/0   \n",
      "21  MODERATE                     BDNF      ENSG00000176697  ...      0/1   \n",
      "22  MODERATE                   LRRC23      ENSG00000010626  ...      0/1   \n",
      "23  MODERATE                    LRRK2      ENSG00000188906  ...      0/0   \n",
      "24  MODERATE                    LRRK2      ENSG00000188906  ...      0/0   \n",
      "25  MODERATE                     NQO1      ENSG00000181019  ...      0/0   \n",
      "26  MODERATE                     GFAP      ENSG00000131095  ...      ./.   \n",
      "27  MODERATE                     MAPT      ENSG00000186868  ...      0/0   \n",
      "28  MODERATE                     INSR      ENSG00000171105  ...      0/1   \n",
      "29  MODERATE                     APOE      ENSG00000130203  ...      0/0   \n",
      "30  MODERATE                     APOE      ENSG00000130203  ...      0/0   \n",
      "\n",
      "   UKCON130 UKCON143 UKCON144 UKCON135 UKCON136 UKCON137  UKCON138  UKCON139  \\\n",
      "0       0/1      0/1      1/1      0/0      0/1      0/1       0/1       0/1   \n",
      "1       0/0      0/0      0/0      0/0      0/1      0/1       1/1       0/1   \n",
      "2       0/1      0/0      0/0      0/1      1/1      0/1       0/1       NaN   \n",
      "3       0/0      0/0      0/0      0/1      0/0      0/0       0/0       0/0   \n",
      "4       0/1      0/1      0/0      0/0      0/0      0/1       0/0       NaN   \n",
      "5       0/1      0/0      0/1      0/0      0/0      0/0       0/0       0/1   \n",
      "6       ./.      0/1      0/0      0/0      0/0      0/0       0/0       0/0   \n",
      "7       0/0      0/0      0/0      0/1      1/1      0/0       0/0       0/0   \n",
      "8       0/0      1/1      0/1      0/1      0/0      0/0       0/0       NaN   \n",
      "9       0/0      0/0      0/0      0/1      0/1      0/0       0/0       NaN   \n",
      "10      0/1      1/1      1/1      1/1      0/0      0/0       0/0       0/1   \n",
      "11      0/0      0/0      0/0      0/1      0/0      0/0       0/0       0/0   \n",
      "12      0/1      ./.      0/0      0/0      0/0      0/0       0/0       0/0   \n",
      "13      0/0      0/0      0/0      0/0      0/0      0/0       0/1       0/0   \n",
      "14      0/1      0/0      0/1      0/0      0/0      0/0       0/1       0/1   \n",
      "15      0/0      0/1      0/0      0/0      0/0      0/0       0/0       0/0   \n",
      "16      0/0      0/1      0/0      0/0      0/0      1/1       0/0       0/1   \n",
      "17      0/0      0/0      0/0      0/1      0/0      0/0       0/0       0/0   \n",
      "18      0/0      0/0      0/1      0/0      0/0      0/0       0/0       NaN   \n",
      "19      0/1      0/0      0/1      0/1      0/0      0/0       0/0       NaN   \n",
      "20      0/0      0/0      0/1      0/1      0/0      0/0       0/0       NaN   \n",
      "21      0/0      0/1      0/1      0/1      0/0      0/1       0/0       1/1   \n",
      "22      0/1      0/0      0/0      0/0      1/1      0/0       0/0       0/1   \n",
      "23      0/1      0/1      0/0      0/0      0/1      0/1       0/1       0/1   \n",
      "24      ./.      0/0      0/0      0/0      0/0      0/0       0/0       0/0   \n",
      "25      0/0      0/0      0/0      0/0      0/0      0/1       0/0       0/0   \n",
      "26      ./.      0/0      0/0      0/0      0/0      0/0       0/0       0/0   \n",
      "27      0/0      0/0      0/1      0/0      0/0      0/1       0/1       0/1   \n",
      "28      0/0      0/1      0/0      0/0      0/0      0/0       0/0       0/0   \n",
      "29      0/0      0/0      0/0      0/0      0/0      0/0       0/0       0/0   \n",
      "30      0/0      0/1      0/0      0/0      0/0      0/0       0/0       0/0   \n",
      "\n",
      "        \n",
      "0  NaN  \n",
      "1  NaN  \n",
      "2  NaN  \n",
      "3  NaN  \n",
      "4  NaN  \n",
      "5  NaN  \n",
      "6  NaN  \n",
      "7  NaN  \n",
      "8  NaN  \n",
      "9  NaN  \n",
      "10 NaN  \n",
      "11 NaN  \n",
      "12 NaN  \n",
      "13 NaN  \n",
      "14 NaN  \n",
      "15 NaN  \n",
      "16 NaN  \n",
      "17 NaN  \n",
      "18 NaN  \n",
      "19 NaN  \n",
      "20 NaN  \n",
      "21 NaN  \n",
      "22 NaN  \n",
      "23 NaN  \n",
      "24 NaN  \n",
      "25 NaN  \n",
      "26 NaN  \n",
      "27 NaN  \n",
      "28 NaN  \n",
      "29 NaN  \n",
      "30 NaN  \n",
      "\n",
      "[31 rows x 1676 columns]\n",
      "Index(['Gene', 'Feature_type', 'Feature', 'BIOTYPE', 'EXON', 'INTRON', 'HGVSc',\n",
      "       'HGVSp', 'cDNA_position', 'CDS_position', 'Protein_position',\n",
      "       'Amino_acids', 'Codons', 'Existing_variation', 'DISTANCE', 'STRAND',\n",
      "       'FLAGS', 'VARIANT_CLASS', 'SYMBOL_SOURCE', 'HGNC_ID', 'CANONICAL',\n",
      "       'MANE_SELECT', 'MANE_PLUS_CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP',\n",
      "       'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT_ISOFORM', 'REFSEQ_MATCH',\n",
      "       'SOURCE', 'REFSEQ_OFFSET', 'GIVEN_REF', 'USED_REF', 'BAM_EDIT',\n",
      "       'GENE_PHENO'],\n",
      "      dtype='object')\n",
      "    CHROM        POS          ID REF ALT Allele           Consequence  \\\n",
      "0       1   11854476   rs1801131   A   C      C      missense_variant   \n",
      "1       1   20977000   rs1043424   A   C      C      missense_variant   \n",
      "2       1  209782343  rs11119314   T   C      C      missense_variant   \n",
      "3       2  210558162    rs741006   G   A      A      missense_variant   \n",
      "4       2  234113301      rs9247   C   T      T      missense_variant   \n",
      "5       3   12393125   rs1801282   C   G      G      missense_variant   \n",
      "6       3  133475722   rs1799852   G   A      A      missense_variant   \n",
      "7       3  133494354   rs1049296   C   T      T      missense_variant   \n",
      "8       4   23815662   rs8192678   C   T      T      missense_variant   \n",
      "9       4   23815681  rs17574213   C   T      T      missense_variant   \n",
      "10      5  179264731     rs10277   G   A      A      missense_variant   \n",
      "11      6   26093141   rs1800562   C   T      T      missense_variant   \n",
      "12      6   52849325   rs1802061   G   A      A      missense_variant   \n",
      "13      6  161781225   rs1801334   G   A      A      missense_variant   \n",
      "14      6  161807855   rs1801582   C   G      G      missense_variant   \n",
      "15      7  143095153  rs11768549   G   A      A      missense_variant   \n",
      "16      7  154587793  rs10236495   A   G      G      missense_variant   \n",
      "17      7  154672646   rs3817522   A   G      G      missense_variant   \n",
      "18      9  135139826   rs3739927   T   C      C      missense_variant   \n",
      "19      9  135172412   rs2296869   T   C      C      missense_variant   \n",
      "20      9  135173685   rs2296871   A   G      G      missense_variant   \n",
      "21     11   27679916      rs6265   C   T      T      missense_variant   \n",
      "22     12    7023123    rs710415   A   G      G      missense_variant   \n",
      "23     12   40631791  rs10878245   G   A      A      missense_variant   \n",
      "24     12   40657700   rs7308720   C   G      G      missense_variant   \n",
      "25     16   69745145   rs1800566   C   T      T      missense_variant   \n",
      "26     17   42987524   rs9916491   A   G      G      missense_variant   \n",
      "27     17   44067382   rs2258689   T   C      C      missense_variant   \n",
      "28     19    7141775   rs2229431   C   T      T      missense_variant   \n",
      "29     19   45411941    rs429358   T   C      C      missense_variant   \n",
      "30     19   45412079      rs7412   C   T      T      missense_variant   \n",
      "\n",
      "      IMPACT                   SYMBOL  \\\n",
      "0   MODERATE                    MTHFR   \n",
      "1   MODERATE                    PINK1   \n",
      "2   MODERATE                   CAMK1G   \n",
      "3   MODERATE                     MAP2   \n",
      "4   MODERATE                   INPP5D   \n",
      "5   MODERATE                    PPARG   \n",
      "6   MODERATE                       TF   \n",
      "7   MODERATE                       TF   \n",
      "8   MODERATE  PPARGC1AENSG00000109819   \n",
      "9   MODERATE  PPARGC1AENSG00000109819   \n",
      "10  MODERATE                  C5orf45   \n",
      "11  MODERATE                      HFE   \n",
      "12  MODERATE                    GSTA4   \n",
      "13  MODERATE                    PARK2   \n",
      "14  MODERATE                    PARK2   \n",
      "15  MODERATE                    EPHA1   \n",
      "16  MODERATE                     DPP6   \n",
      "17  MODERATE                     DPP6   \n",
      "18  MODERATE                     SETX   \n",
      "19  MODERATE                     SETX   \n",
      "20  MODERATE                     SETX   \n",
      "21  MODERATE                     BDNF   \n",
      "22  MODERATE                   LRRC23   \n",
      "23  MODERATE                    LRRK2   \n",
      "24  MODERATE                    LRRK2   \n",
      "25  MODERATE                     NQO1   \n",
      "26  MODERATE                     GFAP   \n",
      "27  MODERATE                     MAPT   \n",
      "28  MODERATE                     INSR   \n",
      "29  MODERATE                     APOE   \n",
      "30  MODERATE                     APOE   \n",
      "\n",
      "                                          SIFT  ... UKCON128 UKCON130  \\\n",
      "0                            deleterious(0.01)  ...      0/1      0/1   \n",
      "1                              tolerated(0.42)  ...      0/0      0/0   \n",
      "2   deleterious_low_confidence(0.01)unknown(0)  ...      0/1      0/1   \n",
      "3                  tolerated_low_confidence(1)  ...      0/0      0/0   \n",
      "4                     possibly_damaging(0.814)  ...      0/1      0/1   \n",
      "5               tolerated_low_confidence(0.05)  ...      0/0      0/1   \n",
      "6                              tolerated(0.11)  ...      0/0      ./.   \n",
      "7                              tolerated(0.13)  ...      0/0      0/0   \n",
      "8                                benign(0.003)  ...      1/1      0/0   \n",
      "9                                benign(0.019)  ...      0/0      0/0   \n",
      "10                           deleterious(0.05)  ...      0/1      0/1   \n",
      "11                           deleterious(0.01)  ...      0/0      0/0   \n",
      "12                              deleterious(0)  ...      0/0      0/1   \n",
      "13                              deleterious(0)  ...      0/0      0/0   \n",
      "14                             tolerated(0.48)  ...      0/0      0/1   \n",
      "15                             tolerated(0.25)  ...      0/0      0/0   \n",
      "16                                           .  ...      1/1      0/0   \n",
      "17                              deleterious(0)  ...      0/0      0/0   \n",
      "18                                   benign(0)  ...      0/0      0/0   \n",
      "19                               benign(0.212)  ...      0/1      0/1   \n",
      "20                               benign(0.012)  ...      0/0      0/0   \n",
      "21                              tolerated(0.2)  ...      0/1      0/0   \n",
      "22              tolerated_low_confidence(0.49)  ...      0/1      0/1   \n",
      "23                           deleterious(0.03)  ...      0/0      0/1   \n",
      "24                           deleterious(0.01)  ...      0/0      ./.   \n",
      "25                              deleterious(0)  ...      0/0      0/0   \n",
      "26                           deleterious(0.01)  ...      ./.      ./.   \n",
      "27                                tolerated(1)  ...      0/0      0/0   \n",
      "28                             tolerated(0.08)  ...      0/1      0/0   \n",
      "29                                tolerated(1)  ...      0/0      0/0   \n",
      "30                              deleterious(0)  ...      0/0      0/0   \n",
      "\n",
      "   UKCON143 UKCON144 UKCON135 UKCON136 UKCON137 UKCON138 UKCON139      \n",
      "0       0/1      1/1      0/0      0/1      0/1      0/1      0/1 NaN  \n",
      "1       0/0      0/0      0/0      0/1      0/1      1/1      0/1 NaN  \n",
      "2       0/0      0/0      0/1      1/1      0/1      0/1      NaN NaN  \n",
      "3       0/0      0/0      0/1      0/0      0/0      0/0      0/0 NaN  \n",
      "4       0/1      0/0      0/0      0/0      0/1      0/0      NaN NaN  \n",
      "5       0/0      0/1      0/0      0/0      0/0      0/0      0/1 NaN  \n",
      "6       0/1      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "7       0/0      0/0      0/1      1/1      0/0      0/0      0/0 NaN  \n",
      "8       1/1      0/1      0/1      0/0      0/0      0/0      NaN NaN  \n",
      "9       0/0      0/0      0/1      0/1      0/0      0/0      NaN NaN  \n",
      "10      1/1      1/1      1/1      0/0      0/0      0/0      0/1 NaN  \n",
      "11      0/0      0/0      0/1      0/0      0/0      0/0      0/0 NaN  \n",
      "12      ./.      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "13      0/0      0/0      0/0      0/0      0/0      0/1      0/0 NaN  \n",
      "14      0/0      0/1      0/0      0/0      0/0      0/1      0/1 NaN  \n",
      "15      0/1      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "16      0/1      0/0      0/0      0/0      1/1      0/0      0/1 NaN  \n",
      "17      0/0      0/0      0/1      0/0      0/0      0/0      0/0 NaN  \n",
      "18      0/0      0/1      0/0      0/0      0/0      0/0      NaN NaN  \n",
      "19      0/0      0/1      0/1      0/0      0/0      0/0      NaN NaN  \n",
      "20      0/0      0/1      0/1      0/0      0/0      0/0      NaN NaN  \n",
      "21      0/1      0/1      0/1      0/0      0/1      0/0      1/1 NaN  \n",
      "22      0/0      0/0      0/0      1/1      0/0      0/0      0/1 NaN  \n",
      "23      0/1      0/0      0/0      0/1      0/1      0/1      0/1 NaN  \n",
      "24      0/0      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "25      0/0      0/0      0/0      0/0      0/1      0/0      0/0 NaN  \n",
      "26      0/0      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "27      0/0      0/1      0/0      0/0      0/1      0/1      0/1 NaN  \n",
      "28      0/1      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "29      0/0      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "30      0/1      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "\n",
      "[31 rows x 1638 columns]\n",
      "    CHROM        POS          ID REF ALT Allele           Consequence  \\\n",
      "0       1   11854476   rs1801131   A   C      C      missense_variant   \n",
      "1       1   20977000   rs1043424   A   C      C      missense_variant   \n",
      "2       1  209782343  rs11119314   T   C      C      missense_variant   \n",
      "3       2  210558162    rs741006   G   A      A      missense_variant   \n",
      "4       2  234113301      rs9247   C   T      T      missense_variant   \n",
      "5       3   12393125   rs1801282   C   G      G      missense_variant   \n",
      "6       3  133475722   rs1799852   G   A      A      missense_variant   \n",
      "7       3  133494354   rs1049296   C   T      T      missense_variant   \n",
      "8       4   23815662   rs8192678   C   T      T      missense_variant   \n",
      "9       4   23815681  rs17574213   C   T      T      missense_variant   \n",
      "10      5  179264731     rs10277   G   A      A      missense_variant   \n",
      "11      6   26093141   rs1800562   C   T      T      missense_variant   \n",
      "12      6   52849325   rs1802061   G   A      A      missense_variant   \n",
      "13      6  161781225   rs1801334   G   A      A      missense_variant   \n",
      "14      6  161807855   rs1801582   C   G      G      missense_variant   \n",
      "15      7  143095153  rs11768549   G   A      A      missense_variant   \n",
      "16      7  154587793  rs10236495   A   G      G      missense_variant   \n",
      "17      7  154672646   rs3817522   A   G      G      missense_variant   \n",
      "18      9  135139826   rs3739927   T   C      C      missense_variant   \n",
      "19      9  135172412   rs2296869   T   C      C      missense_variant   \n",
      "20      9  135173685   rs2296871   A   G      G      missense_variant   \n",
      "21     11   27679916      rs6265   C   T      T      missense_variant   \n",
      "22     12    7023123    rs710415   A   G      G      missense_variant   \n",
      "23     12   40631791  rs10878245   G   A      A      missense_variant   \n",
      "24     12   40657700   rs7308720   C   G      G      missense_variant   \n",
      "25     16   69745145   rs1800566   C   T      T      missense_variant   \n",
      "26     17   42987524   rs9916491   A   G      G      missense_variant   \n",
      "27     17   44067382   rs2258689   T   C      C      missense_variant   \n",
      "28     19    7141775   rs2229431   C   T      T      missense_variant   \n",
      "29     19   45411941    rs429358   T   C      C      missense_variant   \n",
      "30     19   45412079      rs7412   C   T      T      missense_variant   \n",
      "\n",
      "      IMPACT                   SYMBOL  \\\n",
      "0   MODERATE                    MTHFR   \n",
      "1   MODERATE                    PINK1   \n",
      "2   MODERATE                   CAMK1G   \n",
      "3   MODERATE                     MAP2   \n",
      "4   MODERATE                   INPP5D   \n",
      "5   MODERATE                    PPARG   \n",
      "6   MODERATE                       TF   \n",
      "7   MODERATE                       TF   \n",
      "8   MODERATE  PPARGC1AENSG00000109819   \n",
      "9   MODERATE  PPARGC1AENSG00000109819   \n",
      "10  MODERATE                  C5orf45   \n",
      "11  MODERATE                      HFE   \n",
      "12  MODERATE                    GSTA4   \n",
      "13  MODERATE                    PARK2   \n",
      "14  MODERATE                    PARK2   \n",
      "15  MODERATE                    EPHA1   \n",
      "16  MODERATE                     DPP6   \n",
      "17  MODERATE                     DPP6   \n",
      "18  MODERATE                     SETX   \n",
      "19  MODERATE                     SETX   \n",
      "20  MODERATE                     SETX   \n",
      "21  MODERATE                     BDNF   \n",
      "22  MODERATE                   LRRC23   \n",
      "23  MODERATE                    LRRK2   \n",
      "24  MODERATE                    LRRK2   \n",
      "25  MODERATE                     NQO1   \n",
      "26  MODERATE                     GFAP   \n",
      "27  MODERATE                     MAPT   \n",
      "28  MODERATE                     INSR   \n",
      "29  MODERATE                     APOE   \n",
      "30  MODERATE                     APOE   \n",
      "\n",
      "                                          SIFT  ... UKCON128 UKCON130  \\\n",
      "0                            deleterious(0.01)  ...      0/1      0/1   \n",
      "1                              tolerated(0.42)  ...      0/0      0/0   \n",
      "2   deleterious_low_confidence(0.01)unknown(0)  ...      0/1      0/1   \n",
      "3                  tolerated_low_confidence(1)  ...      0/0      0/0   \n",
      "4                     possibly_damaging(0.814)  ...      0/1      0/1   \n",
      "5               tolerated_low_confidence(0.05)  ...      0/0      0/1   \n",
      "6                              tolerated(0.11)  ...      0/0      ./.   \n",
      "7                              tolerated(0.13)  ...      0/0      0/0   \n",
      "8                                benign(0.003)  ...      1/1      0/0   \n",
      "9                                benign(0.019)  ...      0/0      0/0   \n",
      "10                           deleterious(0.05)  ...      0/1      0/1   \n",
      "11                           deleterious(0.01)  ...      0/0      0/0   \n",
      "12                              deleterious(0)  ...      0/0      0/1   \n",
      "13                              deleterious(0)  ...      0/0      0/0   \n",
      "14                             tolerated(0.48)  ...      0/0      0/1   \n",
      "15                             tolerated(0.25)  ...      0/0      0/0   \n",
      "16                                           .  ...      1/1      0/0   \n",
      "17                              deleterious(0)  ...      0/0      0/0   \n",
      "18                                   benign(0)  ...      0/0      0/0   \n",
      "19                               benign(0.212)  ...      0/1      0/1   \n",
      "20                               benign(0.012)  ...      0/0      0/0   \n",
      "21                              tolerated(0.2)  ...      0/1      0/0   \n",
      "22              tolerated_low_confidence(0.49)  ...      0/1      0/1   \n",
      "23                           deleterious(0.03)  ...      0/0      0/1   \n",
      "24                           deleterious(0.01)  ...      0/0      ./.   \n",
      "25                              deleterious(0)  ...      0/0      0/0   \n",
      "26                           deleterious(0.01)  ...      ./.      ./.   \n",
      "27                                tolerated(1)  ...      0/0      0/0   \n",
      "28                             tolerated(0.08)  ...      0/1      0/0   \n",
      "29                                tolerated(1)  ...      0/0      0/0   \n",
      "30                              deleterious(0)  ...      0/0      0/0   \n",
      "\n",
      "   UKCON143 UKCON144 UKCON135 UKCON136 UKCON137 UKCON138 UKCON139      \n",
      "0       0/1      1/1      0/0      0/1      0/1      0/1      0/1 NaN  \n",
      "1       0/0      0/0      0/0      0/1      0/1      1/1      0/1 NaN  \n",
      "2       0/0      0/0      0/1      1/1      0/1      0/1      NaN NaN  \n",
      "3       0/0      0/0      0/1      0/0      0/0      0/0      0/0 NaN  \n",
      "4       0/1      0/0      0/0      0/0      0/1      0/0      NaN NaN  \n",
      "5       0/0      0/1      0/0      0/0      0/0      0/0      0/1 NaN  \n",
      "6       0/1      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "7       0/0      0/0      0/1      1/1      0/0      0/0      0/0 NaN  \n",
      "8       1/1      0/1      0/1      0/0      0/0      0/0      NaN NaN  \n",
      "9       0/0      0/0      0/1      0/1      0/0      0/0      NaN NaN  \n",
      "10      1/1      1/1      1/1      0/0      0/0      0/0      0/1 NaN  \n",
      "11      0/0      0/0      0/1      0/0      0/0      0/0      0/0 NaN  \n",
      "12      ./.      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "13      0/0      0/0      0/0      0/0      0/0      0/1      0/0 NaN  \n",
      "14      0/0      0/1      0/0      0/0      0/0      0/1      0/1 NaN  \n",
      "15      0/1      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "16      0/1      0/0      0/0      0/0      1/1      0/0      0/1 NaN  \n",
      "17      0/0      0/0      0/1      0/0      0/0      0/0      0/0 NaN  \n",
      "18      0/0      0/1      0/0      0/0      0/0      0/0      NaN NaN  \n",
      "19      0/0      0/1      0/1      0/0      0/0      0/0      NaN NaN  \n",
      "20      0/0      0/1      0/1      0/0      0/0      0/0      NaN NaN  \n",
      "21      0/1      0/1      0/1      0/0      0/1      0/0      1/1 NaN  \n",
      "22      0/0      0/0      0/0      1/1      0/0      0/0      0/1 NaN  \n",
      "23      0/1      0/0      0/0      0/1      0/1      0/1      0/1 NaN  \n",
      "24      0/0      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "25      0/0      0/0      0/0      0/0      0/1      0/0      0/0 NaN  \n",
      "26      0/0      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "27      0/0      0/1      0/0      0/0      0/1      0/1      0/1 NaN  \n",
      "28      0/1      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "29      0/0      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "30      0/1      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "\n",
      "[31 rows x 1607 columns]\n",
      "    CHROM        POS          ID REF ALT           Consequence  \\\n",
      "0       1   11854476   rs1801131   A   C      missense_variant   \n",
      "1       1   20977000   rs1043424   A   C      missense_variant   \n",
      "2       1  209782343  rs11119314   T   C      missense_variant   \n",
      "3       2  210558162    rs741006   G   A      missense_variant   \n",
      "4       2  234113301      rs9247   C   T      missense_variant   \n",
      "5       3   12393125   rs1801282   C   G      missense_variant   \n",
      "6       3  133475722   rs1799852   G   A      missense_variant   \n",
      "7       3  133494354   rs1049296   C   T      missense_variant   \n",
      "8       4   23815662   rs8192678   C   T      missense_variant   \n",
      "9       4   23815681  rs17574213   C   T      missense_variant   \n",
      "10      5  179264731     rs10277   G   A      missense_variant   \n",
      "11      6   26093141   rs1800562   C   T      missense_variant   \n",
      "12      6   52849325   rs1802061   G   A      missense_variant   \n",
      "13      6  161781225   rs1801334   G   A      missense_variant   \n",
      "14      6  161807855   rs1801582   C   G      missense_variant   \n",
      "15      7  143095153  rs11768549   G   A      missense_variant   \n",
      "16      7  154587793  rs10236495   A   G      missense_variant   \n",
      "17      7  154672646   rs3817522   A   G      missense_variant   \n",
      "18      9  135139826   rs3739927   T   C      missense_variant   \n",
      "19      9  135172412   rs2296869   T   C      missense_variant   \n",
      "20      9  135173685   rs2296871   A   G      missense_variant   \n",
      "21     11   27679916      rs6265   C   T      missense_variant   \n",
      "22     12    7023123    rs710415   A   G      missense_variant   \n",
      "23     12   40631791  rs10878245   G   A      missense_variant   \n",
      "24     12   40657700   rs7308720   C   G      missense_variant   \n",
      "25     16   69745145   rs1800566   C   T      missense_variant   \n",
      "26     17   42987524   rs9916491   A   G      missense_variant   \n",
      "27     17   44067382   rs2258689   T   C      missense_variant   \n",
      "28     19    7141775   rs2229431   C   T      missense_variant   \n",
      "29     19   45411941    rs429358   T   C      missense_variant   \n",
      "30     19   45412079      rs7412   C   T      missense_variant   \n",
      "\n",
      "                     SYMBOL                                        SIFT  \\\n",
      "0                     MTHFR                           deleterious(0.01)   \n",
      "1                     PINK1                             tolerated(0.42)   \n",
      "2                    CAMK1G  deleterious_low_confidence(0.01)unknown(0)   \n",
      "3                      MAP2                 tolerated_low_confidence(1)   \n",
      "4                    INPP5D                    possibly_damaging(0.814)   \n",
      "5                     PPARG              tolerated_low_confidence(0.05)   \n",
      "6                        TF                             tolerated(0.11)   \n",
      "7                        TF                             tolerated(0.13)   \n",
      "8   PPARGC1AENSG00000109819                               benign(0.003)   \n",
      "9   PPARGC1AENSG00000109819                               benign(0.019)   \n",
      "10                  C5orf45                           deleterious(0.05)   \n",
      "11                      HFE                           deleterious(0.01)   \n",
      "12                    GSTA4                              deleterious(0)   \n",
      "13                    PARK2                              deleterious(0)   \n",
      "14                    PARK2                             tolerated(0.48)   \n",
      "15                    EPHA1                             tolerated(0.25)   \n",
      "16                     DPP6                                           .   \n",
      "17                     DPP6                              deleterious(0)   \n",
      "18                     SETX                                   benign(0)   \n",
      "19                     SETX                               benign(0.212)   \n",
      "20                     SETX                               benign(0.012)   \n",
      "21                     BDNF                              tolerated(0.2)   \n",
      "22                   LRRC23              tolerated_low_confidence(0.49)   \n",
      "23                    LRRK2                           deleterious(0.03)   \n",
      "24                    LRRK2                           deleterious(0.01)   \n",
      "25                     NQO1                              deleterious(0)   \n",
      "26                     GFAP                           deleterious(0.01)   \n",
      "27                     MAPT                                tolerated(1)   \n",
      "28                     INSR                             tolerated(0.08)   \n",
      "29                     APOE                                tolerated(1)   \n",
      "30                     APOE                              deleterious(0)   \n",
      "\n",
      "                                             PolyPhen EURCON119  ... UKCON128  \\\n",
      "0                            possibly_damaging(0.654)       1/1  ...      0/1   \n",
      "1                                       benign(0.006)       0/0  ...      0/0   \n",
      "2                   SMART:SM00220&PROSITE_profiles...       0/1  ...      0/1   \n",
      "3                                           benign(0)       0/0  ...      0/0   \n",
      "4                                                   .       0/0  ...      0/1   \n",
      "5                                           benign(0)       0/0  ...      0/0   \n",
      "6                                       benign(0.096)       0/0  ...      0/0   \n",
      "7                                       benign(0.018)       0/1  ...      0/0   \n",
      "8               PANTHER:PTHR15528&PANTHER:PTHR1552...       0/0  ...      1/1   \n",
      "9            PANTHER:PTHR15528&PANTHER:PTHR15528:SF10       0/0  ...      0/0   \n",
      "10                                      benign(0.001)       0/1  ...      0/1   \n",
      "11                           probably_damaging(0.967)       0/0  ...      0/0   \n",
      "12                           possibly_damaging(0.803)       0/0  ...      0/0   \n",
      "13                           probably_damaging(0.977)       0/0  ...      0/0   \n",
      "14                                          benign(0)       0/1  ...      0/0   \n",
      "15                                      benign(0.044)       0/0  ...      0/0   \n",
      "16                                                  .       0/1  ...      1/1   \n",
      "17                            possibly_damaging(0.83)       0/1  ...      0/0   \n",
      "18          PANTHER:PTHR10887&PANTHER:PTHR10887:SF336       0/0  ...      0/0   \n",
      "19  Gene3D:3.40.50.300&Pfam:PF13086&PANTHER:PTHR10...       0/1  ...      0/1   \n",
      "20          PANTHER:PTHR10887&PANTHER:PTHR10887:SF336       0/1  ...      0/0   \n",
      "21                           probably_damaging(0.991)       0/0  ...      0/1   \n",
      "22                                          benign(0)       0/0  ...      0/1   \n",
      "23                           possibly_damaging(0.704)       0/1  ...      0/0   \n",
      "24                           probably_damaging(0.972)       0/0  ...      0/0   \n",
      "25                           possibly_damaging(0.849)       0/0  ...      0/0   \n",
      "26                                          benign(0)       1/1  ...      ./.   \n",
      "27                                          benign(0)       1/1  ...      0/0   \n",
      "28                                      benign(0.111)       0/0  ...      0/1   \n",
      "29                                      benign(0.001)       0/0  ...      0/0   \n",
      "30                               probably_damaging(1)       0/0  ...      0/0   \n",
      "\n",
      "   UKCON130 UKCON143 UKCON144 UKCON135 UKCON136 UKCON137 UKCON138 UKCON139      \n",
      "0       0/1      0/1      1/1      0/0      0/1      0/1      0/1      0/1 NaN  \n",
      "1       0/0      0/0      0/0      0/0      0/1      0/1      1/1      0/1 NaN  \n",
      "2       0/1      0/0      0/0      0/1      1/1      0/1      0/1      NaN NaN  \n",
      "3       0/0      0/0      0/0      0/1      0/0      0/0      0/0      0/0 NaN  \n",
      "4       0/1      0/1      0/0      0/0      0/0      0/1      0/0      NaN NaN  \n",
      "5       0/1      0/0      0/1      0/0      0/0      0/0      0/0      0/1 NaN  \n",
      "6       ./.      0/1      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "7       0/0      0/0      0/0      0/1      1/1      0/0      0/0      0/0 NaN  \n",
      "8       0/0      1/1      0/1      0/1      0/0      0/0      0/0      NaN NaN  \n",
      "9       0/0      0/0      0/0      0/1      0/1      0/0      0/0      NaN NaN  \n",
      "10      0/1      1/1      1/1      1/1      0/0      0/0      0/0      0/1 NaN  \n",
      "11      0/0      0/0      0/0      0/1      0/0      0/0      0/0      0/0 NaN  \n",
      "12      0/1      ./.      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "13      0/0      0/0      0/0      0/0      0/0      0/0      0/1      0/0 NaN  \n",
      "14      0/1      0/0      0/1      0/0      0/0      0/0      0/1      0/1 NaN  \n",
      "15      0/0      0/1      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "16      0/0      0/1      0/0      0/0      0/0      1/1      0/0      0/1 NaN  \n",
      "17      0/0      0/0      0/0      0/1      0/0      0/0      0/0      0/0 NaN  \n",
      "18      0/0      0/0      0/1      0/0      0/0      0/0      0/0      NaN NaN  \n",
      "19      0/1      0/0      0/1      0/1      0/0      0/0      0/0      NaN NaN  \n",
      "20      0/0      0/0      0/1      0/1      0/0      0/0      0/0      NaN NaN  \n",
      "21      0/0      0/1      0/1      0/1      0/0      0/1      0/0      1/1 NaN  \n",
      "22      0/1      0/0      0/0      0/0      1/1      0/0      0/0      0/1 NaN  \n",
      "23      0/1      0/1      0/0      0/0      0/1      0/1      0/1      0/1 NaN  \n",
      "24      ./.      0/0      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "25      0/0      0/0      0/0      0/0      0/0      0/1      0/0      0/0 NaN  \n",
      "26      ./.      0/0      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "27      0/0      0/0      0/1      0/0      0/0      0/1      0/1      0/1 NaN  \n",
      "28      0/0      0/1      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "29      0/0      0/0      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "30      0/0      0/1      0/0      0/0      0/0      0/0      0/0      0/0 NaN  \n",
      "\n",
      "[31 rows x 1605 columns]\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'exit' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-3-dc511a1c9566>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     15\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     16\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 17\u001b[0;31m     \u001b[0mresult_nodes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcreate_datasets\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcreate_nx_datasets\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmain\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindir\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdataset\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtarget\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdisease\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnetwork\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'missense'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     18\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Coding: number of missense variants per node'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     19\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/CODE/graph_genetics/create_datasets/create_nx_datasets.py\u001b[0m in \u001b[0;36mmain\u001b[0;34m(indir, dataset, target, disease, network, mode, number)\u001b[0m\n\u001b[1;32m    314\u001b[0m         \u001b[0mdiagnosis_file_path\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34mf'{indir}/LOAD/LOAD_metadata.tsv'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    315\u001b[0m         \u001b[0mdiagnosis\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdiagnosis_file_path\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex_col\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 316\u001b[0;31m         \u001b[0mmissense\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mprocess_variants_load\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmissense_file_path\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheader\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdisease\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    317\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    318\u001b[0m     \u001b[0mmissense\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupper\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmissense\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/CODE/graph_genetics/create_datasets/create_nx_datasets.py\u001b[0m in \u001b[0;36mprocess_variants_load\u001b[0;34m(infile, header, disease)\u001b[0m\n\u001b[1;32m     73\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvariants_matrix\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     74\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 75\u001b[0;31m         \u001b[0mexit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     76\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     77\u001b[0m     \u001b[0;31m# Replace genotypes with a numeric value (NaN: miss, 1: presence, 0:absence)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'exit' is not defined"
     ]
    }
   ],
   "source": [
    "dataset = 'LOAD'\n",
    "target  = 'LOAD'\n",
    "diseases = ['AD', 'ND']\n",
    "network = 'original'\n",
    "\n",
    "for disease in diseases:\n",
    "\n",
    "    indir = 'data'\n",
    "    outdir = f'data/graph_datasets/{target}'\n",
    "    print('Input directory:', indir)\n",
    "    print('Output directory:', outdir)\n",
    "    print()\n",
    "\n",
    "    start_time = datetime.datetime.now()\n",
    "    print()\n",
    "\n",
    "    result_nodes = create_datasets.create_nx_datasets.main(indir, dataset, target, disease, network, 'missense', None)\n",
    "    print('Coding: number of missense variants per node')\n",
    "\n",
    "    outfile = f'{outdir}/{disease}_PPI_missense.pkl'\n",
    "    print('Resulting dataset saved at:', outfile)\n",
    "    print()\n",
    "\n",
    "    with open(outfile, 'wb') as f:\n",
    "        pickle.dump(result_nodes, f)\n",
    "\n",
    "    result_nodes_time = datetime.datetime.now()\n",
    "    print('Processing time:', result_nodes_time - start_time)\n",
    "    print('\\n\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**5. Graph classification with GNNs**\n",
    "\n",
    "We then evaluated and tested different GNNs in the framework called [GraphGym](https://github.com/snap-stanford/GraphGym) (You *et al.*, 2020).\n",
    "\n",
    "Configuration and grid files employed are in the subdirectory [graphgym_files](graphgym_files).\n",
    "\n",
    "Summarized results obtained by GraphGym in LOAD dataset are [here](results/GNNs_LOAD/2022_02_LOAD.csv)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**6. Baseline model**\n",
    "\n",
    "Using only APOE gene as input."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Class distribution:\n",
      "1    1014\n",
      "0     585\n",
      "Name: y, dtype: int64\n",
      "1\n",
      "Acc. 0.70625\n",
      "Pre. 0.8674698795180723\n",
      "Rec. 0.6666666666666666\n",
      "F1. 0.7539267015706806\n",
      "AUC. 0.7539267015706806\n",
      "\n",
      "2\n",
      "Acc. 0.6625\n",
      "Pre. 0.7857142857142857\n",
      "Rec. 0.6470588235294118\n",
      "F1. 0.7096774193548386\n",
      "AUC. 0.7096774193548386\n",
      "\n",
      "3\n",
      "Acc. 0.68125\n",
      "Pre. 0.7931034482758621\n",
      "Rec. 0.6764705882352942\n",
      "F1. 0.7301587301587301\n",
      "AUC. 0.7301587301587301\n",
      "\n",
      "4\n",
      "Acc. 0.64375\n",
      "Pre. 0.7848101265822784\n",
      "Rec. 0.6078431372549019\n",
      "F1. 0.6850828729281768\n",
      "AUC. 0.6850828729281768\n",
      "\n",
      "5\n",
      "Acc. 0.65\n",
      "Pre. 0.7319587628865979\n",
      "Rec. 0.7029702970297029\n",
      "F1. 0.7171717171717172\n",
      "AUC. 0.7171717171717172\n",
      "\n",
      "6\n",
      "Acc. 0.65\n",
      "Pre. 0.7047619047619048\n",
      "Rec. 0.7474747474747475\n",
      "F1. 0.7254901960784313\n",
      "AUC. 0.7254901960784313\n",
      "\n",
      "7\n",
      "Acc. 0.59375\n",
      "Pre. 0.75\n",
      "Rec. 0.5887850467289719\n",
      "F1. 0.6596858638743455\n",
      "AUC. 0.6596858638743455\n",
      "\n",
      "8\n",
      "Acc. 0.69375\n",
      "Pre. 0.7311827956989247\n",
      "Rec. 0.7391304347826086\n",
      "F1. 0.7351351351351352\n",
      "AUC. 0.7351351351351352\n",
      "\n",
      "9\n",
      "Acc. 0.7125\n",
      "Pre. 0.7549019607843137\n",
      "Rec. 0.7857142857142857\n",
      "F1. 0.7700000000000001\n",
      "AUC. 0.7700000000000001\n",
      "\n",
      "10\n",
      "Acc. 0.6540880503144654\n",
      "Pre. 0.7857142857142857\n",
      "Rec. 0.6407766990291263\n",
      "F1. 0.7058823529411764\n",
      "AUC. 0.7058823529411764\n",
      "\n",
      "0.6612469530342052 +- 0.03743144474392154\n"
     ]
    }
   ],
   "source": [
    "target = 'LOAD'\n",
    "\n",
    "infile = f'data/table_datasets/AD_PPI_missense_LOAD_labeled.csv'\n",
    "data = pd.read_csv(infile, index_col = 0)\n",
    "data_wclass = create_class_LOAD(data)\n",
    "\n",
    "x = data_wclass.drop(columns=['y'])\n",
    "x = x['APOE']\n",
    "\n",
    "y = data_wclass['y']\n",
    "x.index = x.index.str.upper()\n",
    "\n",
    "baseline_model(x, y)\n",
    "\n",
    "# f = open(f'data/splits/split_{target}.pkl', 'rb')\n",
    "# split_load = pickle.load(f)\n",
    "# f.close()\n",
    "\n",
    "# auc_load = baseline_model(split_load, x, y)\n",
    "# print('Baseline model LOAD, AUC ROC:', auc_load)"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "5e0466fb38f8b8207bee76f1d5f01e5f82e3be8917a80d900a987709dfd81e7a"
  },
  "kernelspec": {
   "display_name": "Python 3.7.10 ('graphgym')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
