{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Main methodology results\n",
    "\n",
    "*Disclaimer*: please note that many parts of this code require the preprocessed data from ADNI (both genetic and diagnostic related) as input. This data has not been uploaded to the repository for privacy reasons."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import pandas as pd\n",
    "import datetime, pickle\n",
    "from create_datasets import create_nx_datasets, create_splits\n",
    "from ml_models.machine_learning_models import create_class_LOAD, baseline_model"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**1. Obtain genes of interest**\n",
    "\n",
    "Using DisGeNET to get Gene-Disease-Associations (GDAs) to Alzheimer's Disease (AD gene set) and other neurodegenerative diseases (ND). This is already obtained from [first part of the results](1_main_methodology.ipynb)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**2. Obtain biological networks**\n",
    "\n",
    "Using genes of interst obtained from DisGeNET, obtain PPI between them from STRING. This is already obtained from [first part of the results](1_main_methodology.ipynb)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**3. Data preprocessing**\n",
    "\n",
    "Please refer to `data_preprocessing` subdirectory for this part.\n",
    "1. [make_BED_files.R](data_preprocessing/make_BED_files.R) creates BED files with the genomic coordinates of the genes of interest. This is already obtained from [first part of the results](1_main_methodology.ipynb).\n",
    "2. [extract_and_annotate_missense_LOAD.sh](data_preprocessing/extract_and_annotate_missense_LOAD.sh) is the script for obtaining missense variants from the VCF files."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**4. Create graph datasets**\n",
    "\n",
    "Create graph datasets (one graph representing each patient) for different targets with ADNI dataset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "data/AD_STRING_PPI_edgelist.txt\n",
      "Network used: AD string\n",
      "# nodes = 59\n",
      "# edges = 115\n",
      "\n",
      "Dataset used: LOAD\n",
      "(11, 1678)\n",
      "(11, 1608)\n",
      "missense\n",
      "(59, 1599)\n",
      "Creating samples graphs...\n",
      "Class: LOAD. Found 1014 positive subjects out of 1599\n",
      "Sample graph used: # nodes = 52 # edges = 111\n",
      "Density = 0.083710407239819 Diameter = 6\n",
      "Coding: number of missense variants per node\n",
      "Resulting dataset saved at: data/graph_datasets/LOAD/AD_PPI_string_missense.pkl\n",
      "\n",
      "Processing time: 0:01:30.601637\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Network used: AD biogrid\n",
      "# nodes = 46\n",
      "# edges = 62\n",
      "\n",
      "Dataset used: LOAD\n",
      "(11, 1678)\n",
      "(11, 1608)\n",
      "missense\n",
      "(46, 1599)\n",
      "Creating samples graphs...\n",
      "Class: LOAD. Found 1014 positive subjects out of 1599\n",
      "Sample graph used: # nodes = 38 # edges = 57\n",
      "Density = 0.08108108108108109 Diameter = 5\n",
      "Coding: number of missense variants per node\n",
      "Resulting dataset saved at: data/graph_datasets/LOAD/AD_PPI_biogrid_missense.pkl\n",
      "\n",
      "Processing time: 0:01:07.566558\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Network used: AD huri\n",
      "# nodes = 18\n",
      "# edges = 16\n",
      "\n",
      "Dataset used: LOAD\n",
      "(11, 1678)\n",
      "(11, 1608)\n",
      "missense\n",
      "(18, 1599)\n",
      "Creating samples graphs...\n",
      "Class: LOAD. Found 1014 positive subjects out of 1599\n",
      "Sample graph used: # nodes = 12 # edges = 13\n",
      "Density = 0.19696969696969696 Diameter = 6\n",
      "Coding: number of missense variants per node\n",
      "Resulting dataset saved at: data/graph_datasets/LOAD/AD_PPI_huri_missense.pkl\n",
      "\n",
      "Processing time: 0:00:27.810840\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Network used: AD snap_brain\n",
      "# nodes = 29\n",
      "# edges = 52\n",
      "\n",
      "Dataset used: LOAD\n",
      "(11, 1678)\n",
      "(11, 1608)\n",
      "missense\n",
      "(29, 1599)\n",
      "Creating samples graphs...\n",
      "Class: LOAD. Found 1014 positive subjects out of 1599\n",
      "Sample graph used: # nodes = 29 # edges = 52\n",
      "Density = 0.12807881773399016 Diameter = 6\n",
      "Coding: number of missense variants per node\n",
      "Resulting dataset saved at: data/graph_datasets/LOAD/AD_PPI_snap_brain_missense.pkl\n",
      "\n",
      "Processing time: 0:00:43.336953\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Network used: AD giant_brain\n",
      "# nodes = 36\n",
      "# edges = 100\n",
      "\n",
      "Dataset used: LOAD\n",
      "(11, 1678)\n",
      "(11, 1608)\n",
      "missense\n",
      "(36, 1599)\n",
      "Creating samples graphs...\n",
      "Class: LOAD. Found 1014 positive subjects out of 1599\n",
      "Sample graph used: # nodes = 36 # edges = 100\n",
      "Density = 0.15873015873015872 Diameter = 6\n",
      "Coding: number of missense variants per node\n",
      "Resulting dataset saved at: data/graph_datasets/LOAD/AD_PPI_giant_brain_missense.pkl\n",
      "\n",
      "Processing time: 0:00:58.647643\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "networks = ['string', 'biogrid', 'huri', 'snap_brain', 'giant_brain']\n",
    "\n",
    "for network in networks:\n",
    "\n",
    "    outdir = f'data/graph_datasets/LOAD'\n",
    "\n",
    "    start_time = datetime.datetime.now()\n",
    "    print()\n",
    "\n",
    "    result_nodes = create_nx_datasets.main('data', 'LOAD', 'LOAD', 'AD', network, 'missense', None)\n",
    "    print('Coding: number of missense variants per node')\n",
    "\n",
    "    outfile = f'{outdir}/AD_PPI_{network}_missense.pkl'\n",
    "    print('Resulting dataset saved at:', outfile)\n",
    "    print()\n",
    "\n",
    "    with open(outfile, 'wb') as f:\n",
    "        pickle.dump(result_nodes, f)\n",
    "\n",
    "    result_nodes_time = datetime.datetime.now()\n",
    "    print('Processing time:', result_nodes_time - start_time)\n",
    "    print('\\n\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LOAD\n",
      "Fold -  1   |   train -  [526 913]   |   test -  [ 59 101]\n",
      "Fold -  2   |   train -  [526 913]   |   test -  [ 59 101]\n",
      "Fold -  3   |   train -  [526 913]   |   test -  [ 59 101]\n",
      "Fold -  4   |   train -  [526 913]   |   test -  [ 59 101]\n",
      "Fold -  5   |   train -  [526 913]   |   test -  [ 59 101]\n",
      "Fold -  6   |   train -  [527 912]   |   test -  [ 58 102]\n",
      "Fold -  7   |   train -  [527 912]   |   test -  [ 58 102]\n",
      "Fold -  8   |   train -  [527 912]   |   test -  [ 58 102]\n",
      "Fold -  9   |   train -  [527 912]   |   test -  [ 58 102]\n",
      "Fold -  10   |   train -  [527 913]   |   test -  [ 58 101]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Create split for using it in GraphGym and non-GNN models\n",
    "# Splits are obtained through 10-Fold Stratied Cross Validation\n",
    "\n",
    "create_splits.create_folds_stratified_cv('LOAD', 10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**5. Graph classification with GNNs**\n",
    "\n",
    "We then evaluated and tested different GNNs in the framework called [GraphGym](https://github.com/snap-stanford/GraphGym) (You *et al.*, 2020).\n",
    "\n",
    "Configuration and grid files employed are in the subdirectory [graphgym_files](graphgym_files).\n",
    "\n",
    "Summarized results obtained by GraphGym in LOAD dataset are [here](results/GNNs_LOAD/2022_02_LOAD.csv)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**6. Baseline model**\n",
    "\n",
    "Using only APOE gene as input."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Class distribution:\n",
      "1    1014\n",
      "0     585\n",
      "Name: y, dtype: int64\n",
      "1\n",
      "Acc. 0.70625\n",
      "Pre. 0.8674698795180723\n",
      "Rec. 0.6666666666666666\n",
      "F1. 0.7539267015706806\n",
      "AUC. 0.7539267015706806\n",
      "\n",
      "2\n",
      "Acc. 0.6625\n",
      "Pre. 0.7857142857142857\n",
      "Rec. 0.6470588235294118\n",
      "F1. 0.7096774193548386\n",
      "AUC. 0.7096774193548386\n",
      "\n",
      "3\n",
      "Acc. 0.68125\n",
      "Pre. 0.7931034482758621\n",
      "Rec. 0.6764705882352942\n",
      "F1. 0.7301587301587301\n",
      "AUC. 0.7301587301587301\n",
      "\n",
      "4\n",
      "Acc. 0.64375\n",
      "Pre. 0.7848101265822784\n",
      "Rec. 0.6078431372549019\n",
      "F1. 0.6850828729281768\n",
      "AUC. 0.6850828729281768\n",
      "\n",
      "5\n",
      "Acc. 0.65\n",
      "Pre. 0.7319587628865979\n",
      "Rec. 0.7029702970297029\n",
      "F1. 0.7171717171717172\n",
      "AUC. 0.7171717171717172\n",
      "\n",
      "6\n",
      "Acc. 0.65\n",
      "Pre. 0.7047619047619048\n",
      "Rec. 0.7474747474747475\n",
      "F1. 0.7254901960784313\n",
      "AUC. 0.7254901960784313\n",
      "\n",
      "7\n",
      "Acc. 0.59375\n",
      "Pre. 0.75\n",
      "Rec. 0.5887850467289719\n",
      "F1. 0.6596858638743455\n",
      "AUC. 0.6596858638743455\n",
      "\n",
      "8\n",
      "Acc. 0.69375\n",
      "Pre. 0.7311827956989247\n",
      "Rec. 0.7391304347826086\n",
      "F1. 0.7351351351351352\n",
      "AUC. 0.7351351351351352\n",
      "\n",
      "9\n",
      "Acc. 0.7125\n",
      "Pre. 0.7549019607843137\n",
      "Rec. 0.7857142857142857\n",
      "F1. 0.7700000000000001\n",
      "AUC. 0.7700000000000001\n",
      "\n",
      "10\n",
      "Acc. 0.6540880503144654\n",
      "Pre. 0.7857142857142857\n",
      "Rec. 0.6407766990291263\n",
      "F1. 0.7058823529411764\n",
      "AUC. 0.7058823529411764\n",
      "\n",
      "0.6612469530342052 +- 0.03743144474392154\n"
     ]
    }
   ],
   "source": [
    "target = 'LOAD'\n",
    "\n",
    "infile = f'data/table_datasets/AD_PPI_missense_LOAD_labeled.csv'\n",
    "data = pd.read_csv(infile, index_col = 0)\n",
    "data_wclass = create_class_LOAD(data)\n",
    "\n",
    "x = data_wclass.drop(columns=['y'])\n",
    "x = x['APOE']\n",
    "\n",
    "y = data_wclass['y']\n",
    "x.index = x.index.str.upper()\n",
    "\n",
    "baseline_model(x, y)\n",
    "\n",
    "# f = open(f'data/splits/split_{target}.pkl', 'rb')\n",
    "# split_load = pickle.load(f)\n",
    "# f.close()\n",
    "\n",
    "# auc_load = baseline_model(split_load, x, y)\n",
    "# print('Baseline model LOAD, AUC ROC:', auc_load)"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "5e0466fb38f8b8207bee76f1d5f01e5f82e3be8917a80d900a987709dfd81e7a"
  },
  "kernelspec": {
   "display_name": "Python 3.7.10 ('graphgym')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
